### PR TITLE
Fixes #8005 - Convert allowed NIC types to strings

### PR DESCRIPTION
**Pull request by ares**
@tstrachota could you please check? If you use zeus you can reproduce the autoloading issue by running

```
zeus test test/functional/unattended_controller_test.rb
```

### CODE REVIEW

File: app/models/nic/base.rb [hunk 1]
```diff
@@ -82,8 +82,9 @@ def require_host?
     end
   end
 
-  require_dependency 'nic/interface'
-
-  TYPES = [ Nic::Managed, Nic::Bond, Nic::BMC ]
+  # types that use can create either using API or UI
+  TYPES = %w(Nic::Managed Nic::Bond Nic::BMC)
```
Comment (ohadlevy) [inline app/models/nic/base.rb:86]: does it make sense not to expose "NIC::.." in the API, and rather pass type = bond, managed etc?

Comment (ares) [inline app/models/nic/base.rb:86]: that would be nice, but we should change it also in display actions, @tstrachota can we add some mapping easily? e.g. Nic::Managed is "Interface" in WebUI

Comment (tstrachota) [inline app/models/nic/base.rb:86]: I'll add the mapping since this fixes an issue I introduced in https://github.com/theforeman/foreman/pull/1848

**Comment by tstrachota** [comment]
There's a replacement for this PR in https://github.com/theforeman/foreman/pull/1871 @ares

**Comment by ares** [comment]
Closing in favour of it, thanks!

**Pull request closed without merging**